pointcloud/basic_octree_utils.go

```diff
@@ -165,11 +161,41 @@ func (octree *basicOctree) helperIterate(lowerBound, upperBound, idx int, fn fun
 			idx += numPoints
 		}
 
-	case LeafNodeFilled:
+	case leafNodeFilled:
 		ok = fn(octree.node.point.P, octree.node.point.D)
 
-	case LeafNodeEmpty:
+	case leafNodeEmpty:
 	}
 
 	return ok
 }
+
+// ConvertPointCloudToBasicOctree converts a pointcloud into a octree pointcloud representation.
+// TBD, add test case
+func ConvertPointCloudToBasicOctree(cloud PointCloud) (PointCloud, error) {
+	center := r3.Vector{
```

Comment:
Given how frequently we compute `center` this, should `Center()` just be a method on `PointCloud` or an attribute within `cloud.MetaData`? At a minimum, could we create a helper function to do this computation so its not 4 lines every time we need to get the center of a pointcloud?

Reply:
my thoughts exactly!